src/software/simulated_tests/validation_functions/robot_colliding_validation.cpp

```diff
@@ -0,0 +1,55 @@
+#include "software/new_geom/line.h"
+#include "software/new_geom/util/intersects.h"
+#include "software/new_geom/util/projection.h"
+#include "software/simulated_tests/validation_functions/touch_ball_in_enemy_defense_validation.h"
+
+void robotCollidedTooFast(std::shared_ptr<World> world_ptr,
+                          ValidationCoroutine::push_type &yield)
+{
+    auto robot_touched_ball = [](std::shared_ptr<World> world_ptr) {
+        std::vector<Robot> friendly_robots = world_ptr->friendlyTeam().getAllRobots();
+        std::vector<Robot> enemy_robots    = world_ptr->enemyTeam().getAllRobots();
+        std::vector<Robot> all_robots;
+        all_robots.reserve(friendly_robots.size() + enemy_robots.size());
+        all_robots.insert(all_robots.end(), friendly_robots.begin(),
+                          friendly_robots.end());
+        all_robots.insert(all_robots.end(), enemy_robots.begin(), enemy_robots.end());
+
+        for (const auto &robot : all_robots)
+        {
+            Point robot_position  = robot.position();
+            Vector robot_velocity = robot.velocity();
+            Circle robot_aabb     = Circle(robot_position, ROBOT_MAX_RADIUS_METERS);
+            for (const auto &other_robot : all_robots)
+            {
+                if (other_robot.id() == robot.id())
```

Comment:
IDs may conflict between enemy and friendly teams, so this wont' catch collisions between a friendly and enemy robot with the same ID

Edit: For pairwise iteration it's probably easier and more readable to just do a nested loop with indices or iterators ([like this](https://codereview.stackexchange.com/a/23287))